--- sandbox/linux/services/libc_interceptor.orig	2018-05-20 14:26:50.042454108 +0200
+++ sandbox/linux/services/libc_interceptor.cc	2018-05-20 14:36:44.669915903 +0200
@@ -212,10 +212,13 @@
     // Nvidia's libGL.so overrides dlsym for an unknown reason and replaces
     // it with a version which doesn't work. In this case we'll get a NULL
     // result. There's not a lot we can do at this point, so we just bodge it!
-    LOG(ERROR) << "Your system is broken: dlsym doesn't work! This has been "
+    // Don't use LOG(ERROR) here as InitLibcLocaltimeFunctions is then called
+    // again causing a pthread_once deadlock.
+    fputs ("Your system is broken: dlsym doesn't work! This has been "
                   "reported to be caused by Nvidia's libGL. You should expect"
                   " time related functions to misbehave. "
-                  "http://code.google.com/p/chromium/issues/detail?id=16800";
+                  "http://code.google.com/p/chromium/issues/detail?id=16800\n",
+                  stderr);
   }
 
   if (!g_libc_funcs->localtime)
